### ISSUE TITLE
[5.x]: Using 2 instances of the same matrix field on an entry type will cause entries to be duplicated across both fields

### ISSUE THREAD

**Issue by brionwolf** [craftcms/cms#14352]
### Description

Using 2 instances of the same matrix field on an entry type will cause entries to be duplicated across both fields

### Steps to reproduce

1. Create a matrix field (ie: `matrix - links - single`).
2. Add the matrix field to a "parent" entry type layout twice, and update the handles of each so they're unique. _(ie: `matrixLinkSource`, and `matrixLinkCTA`)_
3. Create an entry of the Parent's Entry type, and add a block to one of the matrix field instances.

### Expected behavior

- The matrix field instance where the block was added has a new block entry, while the second instance of the matrix field remains empty.

### Actual behavior

- Both instances get the same block and associated content for that block. 
- **Example 1:** An entry with a "Source" link matrix block, and a Call to action "Link" block using the same matrix field, but with a unique handle

![Animated gif of saving an entry with 2 instances of the same matrix field](https://github.com/craftcms/cms/assets/5532234/b1d3ed45-cf5c-4d91-8152-be404ba818e1)

- **Example 2:** Deleting the entry block of one, will delete the block on the other field as well.

![Animated gif of deleting blocks on an matrix field where 2 instances of the same matrix field exist](https://github.com/craftcms/cms/assets/5532234/cd61a464-61b3-4cb9-ac9b-66cfc00bfbb2)

### Craft CMS version

Craft Pro 5.0.0-beta.1

### PHP version

8.3.1

### Operating system and version

MacOS Sonoma 14.3 (23D56)

### Database type and version

MySQL 8.2.0

### Image driver and version

GD 8.3.1

### Installed plugins and versions

- CKEditor | 4.0.x-dev

**Comment by brandonkelly** [comment]
Matrix fields are not multi-instance, and it shouldn’t have been possible to place it in the field layout twice to begin with. Do you know how you were able to accomplish that?

**Comment by brionwolf** [comment]
Thanks for the detail! It looks like I was able to do this when I changed the handle of the matrix field when assigning it to an entry type:

1. Create matrix field
2. Add it to an layout
3. Change the handle name of the matrix field.
4. Save Changes
5. Find the same matrix field in the list of fields
6. Add it again.

![animated gif of adding duplicate matrix fields to a single field layout](https://github.com/craftcms/cms/assets/5532234/5021ffe1-29d0-47d4-bcad-f84188d65ffa)
